rfcs/toast-component.md

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import {Toast} from 'baseui/toast';
+
+export default () => {
+  return (
+    <React.Fragment>
+      <Toast
+        onClose={()=>{console.log('Toast notification was dismissed.')}}
+        autoHideDuration={5000}
+        overrides={{
+          Bosy: props => <CustomContainer>{props.children}</CustomContainer>,
+        }}
+      >
+      This a dismissable info notification that is hidden after 5 seconds automatically.
+      </Toast>
+      <Toast
+        closable={false}
+        kind={KIND.error}
+      >
+      {({dismiss}) => {
+        return (
+          <div>
+            This is an error notification with an action button
+            <button onClick={() => {
+              dismiss();
+              console.log('Action button clicked');
+            }}>
+              Take action
+            </button>
+          </div>
+        );
+      }}
+      </Toast>
+    </React.Fragment>
+  );
+}
+```
+
+## Exports
+
+* `Toast`
+* `StyledRoot`
+* `StyledBody`
+* `StyledCloseButton`
+* `KIND`
+* `PLACEMENT`
+
+## `Toast` API
+
+* `children: React.node | ({dismiss}) => React.node` - Required
+  Toast notification content. The children-as-function receives a `dissmiss` method that can be called to dismiss the notification and can be used as a handler for an action inside the toast content
+* `closable: boolean` - Optional. Defaults to true
+  When set to true a close button is displayed and the notification can be dismissed by a user
+* `kind: 'info' | 'success' | 'warning' | 'error'` - Optional. Defaults to 'info'
+  Defines the type of notification
+* `onClose: function` - Optional
+  A callback function called when a notification is dismissed
+* `overrides: {Root, Body, CloseButton}` - Optional
+  Overrides for presentational components. See "Presentational Components Props API" below.
+  * `[ComponentName]: ReactComponent | {props: {}, style: {}, component: ReactComponent}` - Optional
+* `placement: 'inline' | 'topLeft' | 'top' | 'topRight' | 'bottomRight' | 'bottom' | 'bottomLeft'` - Optional. Defaults to 'top'
+  Position of a toast notification relative to the browser window. If set to 'inline' a notitication is positioned according to a normal flow of the document
+* `autoHideDuration: number` - Optional. Defaults to 0
+  The number of milliseconds to wait before automatically dismissing a notification. This behavior is disabled when the value is set to 0
+
+## Presentational components props API
+
+These properties are passed to every presentational (styled) component that is exported:
+
+* `$kind: type`
+* `$theme: theme`
+
+## KIND Constant
+
+* `info` - Generally used for messages with an informational context
+* `success` - Generally used as a confirmation of a succesful action or operation  
+* `warning` - Generally used for messages with an warning context
+* `error` - Generally used as a notification of an error happened as a result of an action or operation
+
+## PLACEMENT Constant
```

Comment:
`Dock`?

Reply:
Not sure what you mean by `Dock`, could you clarify?

Reply:
Let's keep it in sync with the prop name